native/acter/src/api/common.rs

```diff
@@ -188,6 +193,56 @@ pub enum MsgContent {
         admin_contact: Option<String>,
         limit_type: Option<LimitType>,
     },
+    ProfileChange {
+        user_id: OwnedUserId,
+        display_name_change: Option<Change<Option<String>>>,
+        avatar_url_change: Option<Change<Option<OwnedMxcUri>>>,
+    },
+    MembershipChange {
+        user_id: OwnedUserId,
+        change: Option<String>,
+    },
+}
+
+impl From<&MemberProfileChange> for MsgContent {
+    fn from(value: &MemberProfileChange) -> Self {
+        MsgContent::ProfileChange {
+            user_id: value.user_id().to_owned(),
+            display_name_change: value
+                .displayname_change()
+                .and_then(|c| Change::new(c.new.clone(), c.old.clone())),
+            avatar_url_change: value
+                .avatar_url_change()
+                .and_then(|c| Change::new(c.new.clone(), c.old.clone())),
+        }
+    }
+}
+
+impl From<&RoomMembershipChange> for MsgContent {
```

Comment:
assuming nothing much has changed here.

Reply:
Moved it into `/native/core/src/models/status/membership.rs`, because activity is implemented in `core` module and we have to support space state event for activity.